fitbenchmarking/parsing/tests/test_parsers.py

```diff
@@ -324,6 +359,88 @@ def test_jacobian_evaluation(self, file_format, evaluations_file):
                     actual = fitting_problem.jacobian(x, r[1])
                     assert np.isclose(actual, r[2]).all()
 
+    def test_sparsej_evaluation(self, file_format, evaluations_file):
+        """
+        Test that the sparse Jacobian evaluation is consistent with what
+        would be expected by comparing to some precomputed values with
+        fixed params and x values.
+
+        :param file_format: The name of the file format
+        :type file_format: string
+        :param evaluations_file: Path to a json file containing tests and
+                                 results
+                                 in the following format:
+                                 {"test_file1": [[x1, params1, results1],
+                                                 [x2, params2, results2],
+                                                  ...],
+                                  "test_file2": ...}
+        :type evaluations_file: string
+        """
+        # Note that this test is optional so will only run if the file_format
+        # is added to the SPARSE_JACOBIAN_ENABLED_PARSERS list.
+        if file_format in SPARSE_JACOBIAN_ENABLED_PARSERS:
+            message = 'No function evaluations provided to test ' \
+                f'against for {file_format}'
+            assert (evaluations_file is not None), message
+
+            with open(evaluations_file, 'r') as ef:
+                results = load(ef)
+
+            format_dir = os.path.dirname(evaluations_file)
+
+            for f, tests in results.items():
+                f = os.path.join(format_dir, f)
+
+                parser = ParserFactory.create_parser(f)
+                with parser(f, OPTIONS) as p:
+                    fitting_problem = p.parse()
+
+                for r in tests:
+                    x = np.array(r[0])
+                    actual = fitting_problem.sparse_jacobian(x, r[1])
+                    assert issparse(actual)
+                    assert np.isclose(actual.todense(), r[2]).all()
+
+    def test_mantid_jac_when_no_func_by_user(self, file_format,
```

Comment:
I tried to put this test in another class but now the test fails with an error saying "No module named 'mantid'" which arises when the parser is being created, and I don't understand why, because it was passing when using pytest. It raises an error also if I try to add "import mantid" at the top of the test file

Reply:
Thanks! It's passing now